pkgs/servers/typedb/default.nix

```diff
@@ -0,0 +1,125 @@
+{ stdenv, lib, openjdk, which, system, fetchzip, fetchurl, callPackage
+, server-data ? "~/.typedb_home", server-port ? "1729"
+, server-logs ? "~/.typedb_home/logs" }:
+let
+  typedbVersion = "2.1.1";
+  systems = {
+    x86_64-linux = {
+      src = fetchurl {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+        sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+      };
+      dir = "typedb-all-linux-${typedbVersion}";
+    };
+    x86_64-darwin = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+        sha256 = "1njvssv9h0m13jz0ims96nfl1fjyk7lv8jdb6apcqpbkw7ykz8h8";
+      };
+      dir = "typedb-all-mac-${typedbVersion}";
+    };
+    x86_64-windows = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
```

Comment:
```suggestion
  systems = rec {
    x86_64-linux = {
      src = fetchurl {
        url = "https://github.com/vaticle/typedb/releases/download/${version}/typedb-all-linux-${version}.tar.gz";
        sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
      };
      dir = "typedb-all-linux-${typedbVersion}";
    };
    x86_64-darwin = {
      src = fetchzip {
        url = "https://github.com/vaticle/typedb/releases/download/${version}/typedb-all-mac-${version}.zip";
        sha256 = "1njvssv9h0m13jz0ims96nfl1fjyk7lv8jdb6apcqpbkw7ykz8h8";
      };
      dir = "typedb-all-mac-${typedbVersion}";
    };
    x86_64-windows = {
      src = fetchzip {
        url = "https://github.com/vaticle/typedb/releases/download/${version}/typedb-all-windows-${version}.zip";
```

Reply:
@SuperSandro2000 No need for `rec`. You only introduce `version` references, which don't come from this attrset.